src/Algebra/Morphism/Construct/Composition.agda

```diff
@@ -0,0 +1,404 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The composition of morphisms between algebraic structures.
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --without-K #-}
+
+module Algebra.Morphism.Construct.Composition where
+
+open import Algebra.Bundles
+open import Algebra.Morphism.Structures
+open import Data.Product
+open import Function.Base using (_∘_)
+open import Level using (Level)
+open import Relation.Binary.Morphism.Construct.Composition
+
+private
+  variable
+    a b c ℓ₁ ℓ₂ ℓ₃ : Level
+
+module _
+  {M₁ : Magma a ℓ₁} {M₂ : Magma b ℓ₂} {M₃ : Magma c ℓ₃}
```

Comment:
I don't think you need the full `Magma` structures right? You should be able to make do with only `RawMagma`s + the transitivity of the 3rd equality? The same goes for `Monoid` etc.

I say this because one of the main usages of these modules is proving that e.g. a given `RawGroup` is a full `Group` because it's monomorphic with another `Group`.

Reply:
Should the same be done for Identity? I think that just needs `refl`

Reply:
Yes, you should avoid using the full bundles in all similar code!